### ISSUE TITLE
Frame on Windows For Native Menu

### ISSUE THREAD

**Issue by rsispal** [Seo-Rii/electron-acrylic-window#16]
Hi,

First - thank you for producing this package. So glad that this finally is available for Electron.

I had a question regarding the usage of `frame` on the Electron BrowserWindow. Whilst your documentation states to switch it off, I wondered why as it is actually available (albeit with some styling issues which I explain below - hopefully something fixable):

I really am looking for a way to have a BrowserWindow with the standard Windows menu toolbar along the top (API parity for the Electron Menu). 

Okay. So:

# 1. When `frame: true`
The menu does appear, and the window shows the thick frame (thickFrame: true is default on BrowserWindow)
![Screenshot 2020-08-17 at 14 46 02](https://user-images.githubusercontent.com/23012636/90403111-61484b00-e098-11ea-8017-e015b567e057.png)

# 2. When `frame: true, thickFrame: false`
This removes the thick frame, which then causes a gap around the menu toolbar
![Screenshot 2020-08-17 at 14 51 59](https://user-images.githubusercontent.com/23012636/90404117-d49e8c80-e099-11ea-917e-3cc7515f3d8c.png)

# 2.1. Toggling Windows Light/Dark Appearance
This seems to cause the menu to "fit" the window again.
![Screenshot 2020-08-17 at 14 57 05](https://user-images.githubusercontent.com/23012636/90404389-41198b80-e09a-11ea-9596-9f074f3fa2c1.png)

# Moving Forward
Clearly when the appearance mode is changed, the window undergoes some kind of "refresh" that makes the native menu fit the window. Not too sure how to trace what this process is, but it'd be great if the BrowserWindow could somehow replicate this on startup so your library can offer a BrowserWindow with a menu in line with non-acrylic windows. I have followed your recent PR to get this package into Electron, it'd be great to find a way to fix this minor issue.

I thought I would check your thoughts on this matter as I assume you are probably already aware of this during your development efforts.

**Comment by Seo-Rii** [comment]
I'm sorry, but this is the first time I've noticed this. That's a great idea.
However, when I look at this article and test it, there are a few problems: the first is a minor problem: there is still a slight difference if you change the theme and resize it. The second is a serious problem: the window that has been processed like this cannot be resized.
Of course, if I look at the internal structure of the electron, I'll be able to solve this problem, but I don't have the skills and time to do it :( 
Instead, try custom-electron-titlebar. I haven't tested it yet, but I think it will work well with this module. I'm also considering adding this module to dependencies after testing later.

**Comment by rsispal** [comment]
Hi @Seo-Rii, thanks for your detailed response! 

> the first is a minor problem: there is still a slight difference if you change the theme and resize it. 
I notice the title bar always remains white unlike the menu bar. 

> The second is a serious problem: the window that has been processed like this cannot be resized.
Actually, this is just because of the `thickFrame` property and I can confirm that this happens on the standard BrowserWindow. When `thickFrame: true` you are able to resize the window. When `thickFrame: false` you cannot.

> Instead, try custom-electron-titlebar. I haven't tested it yet, but I think it will work well with this module. I'm also considering adding this module to dependencies after testing later.
I did have a look at this and have tried it before, but it's a little concerning that they are archiving the project shortly.


To summarise, these are the things I noticed that might be fixable. I will research further and see if I can help:
- Title bar does not change colour when the appearance is changed to light/dark
- When `thickFrame: true` it seems like the vibrancy effect is being added to this layer where I think the window shadow should be (no shadow can be seen)

**Comment by rsispal** [comment]
Hi @Seo-Rii, thanks for your detailed response!

---
> the first is a minor problem: there is still a slight difference if you change the theme and resize it. 

I notice the title bar always remains white unlike the menu bar. Are you seeing anything else that I have missed?

---
> The second is a serious problem: the window that has been processed like this cannot be resized.

This appears to be due to the `thickFrame` property and I can confirm that this happens on the standard BrowserWindow. When `thickFrame: true` you are able to resize the window. When `thickFrame: false` you cannot.

---
> Instead, try custom-electron-titlebar. I haven't tested it yet, but I think it will work well with this module. I'm also considering adding this module to dependencies after testing later.

I did have a look at this and have tried it before, but I am hesitant given that they are archiving the project shortly.

---

Whilst I have used NAPI, I am completely new to the Windows accent policy stuff, but I will research and see if I can offer any help. 

The only "real" issue I have noticed is:
- the title bar not changing colour when the appearance changes
- when `thickFrame: false` the menu does not fit to the screen until the window repaints

**Comment by Seo-Rii** [comment]
I'm so glad you can help!
First of all, the color of the title bar does not change only when this module is applied, but it is a problem that appears in all electronic programs. Perhaps to solve the problem, it will be necessary to use the above-mentioned modules, or to create new modules to play that role.
I'll look for a problem that I can't adjust the size when the thickFrame is false.

**Comment by tscpp** [comment]
### Windows 10 control buttons redesigned in electron

**x16 windows 10 control buttons:** [![minimize16.png](https://user-images.githubusercontent.com/47453366/97080882-f585df80-15fe-11eb-8f53-1a280a249426.png)](https://user-images.githubusercontent.com/47453366/97080882-f585df80-15fe-11eb-8f53-1a280a249426.png) [![restore16.png](https://user-images.githubusercontent.com/47453366/97082043-fa4e9180-1606-11eb-9a3a-387c18aad2c9.png)](https://user-images.githubusercontent.com/47453366/97082043-fa4e9180-1606-11eb-9a3a-387c18aad2c9.png) [![maximize16.png](https://user-images.githubusercontent.com/47453366/97080879-f3238580-15fe-11eb-8796-9b38959abbb0.png)](https://user-images.githubusercontent.com/47453366/97080879-f3238580-15fe-11eb-8796-9b38959abbb0.png) [![close16.png](https://user-images.githubusercontent.com/47453366/97080876-f0c12b80-15fe-11eb-83fc-78ef5f3b1cfc.png)](https://user-images.githubusercontent.com/47453366/97080876-f0c12b80-15fe-11eb-83fc-78ef5f3b1cfc.png)

I got the icons by redesigning them in [paint.net](https://getpaint.net). They are precisely identical to the one used in both FileExplorer and Notepad. It can be used on any color or surface due to the transparency in the icons. 

There a 165ms transition on the background of the icon. I calculated this with the frames of a 60fps video and a 30fps video of the transition. I checked on both Notepad and FileExplorer.

The color of the close button is `#E81123`.  Checked on both Notepad and FileExplorer.

The color added when hovering is `rgba(0, 0, 0, 26)`. I got the opacity added to the other control buttons on hover by calculating the difference between the white background (255) and hovered (229). I checked on both Notepad and FileExplorer.

The buttons should be 45x29px but can be designed differently on different applications. In paint.net, the controls are 25x21px.
